### PR TITLE
Cinnamon 5.6

### PR DESCRIPTION
**Pull request by bobby285271**
**Update:**
Please also read https://www.linuxmint.com/rel_vera_cinnamon_whatsnew.php

---

This PR updates Cinnamon to 5.6, which is part of Linux Mint 21.1.

#### Lots of cosmetic changes!

Cosmetic changes are "design changes" (e.g. what theme, color, etc to use). Those changes are made by Linux Mint and we don't do random modifications, report to upstream if you don't like the changes. For items marked with *, you are only affected if you didn't set the related keys explicitly before and you can go back to old settings by using `gsettings`.

The following change set are generated from mint-artwork 1.6.5, cinnamon-common 5.6.2, nemo 5.6.0, mint-y-icons 1.6.3, things can still change in follow-up updates.

- The default cursor theme is now a mint-modified Bibata-Modern-Classic (see [this issue](https://github.com/ful1e5/Bibata_Cursor/issues?q=Linux+Mint%2C+sponsoring+and+pull+requests)), provided by mint-artwork. See change log from [mint-artwork](http://packages.linuxmint.com/pool/main/m/mint-artwork/) (download the `*.tar.xz` file and go to `debian/` folder) for more info. Note that we are removing `vanilla-dmz` from the module.
- \* The default GTK theme and icon theme is now Mint-Y-Aqua, it is changed by mint-artwork.
- \* Home/trash/network desktop icons are no longer presented on desktop, this is announced in [Sep news](https://blog.linuxmint.com/?p=4408), still changed by mint-artwork.
- Nemo no longer highlight selected icons, this is announced in [Oct news](https://blog.linuxmint.com/?p=4424), changed by https://github.com/linuxmint/nemo/commit/fdd47db7a7399ae12e21912fa33bd81cd1ba0eac.
- \* There is a new cornerbar applet that replaces show-desktop, changed by https://github.com/linuxmint/cinnamon/commit/cd47f9aef5c383a108c5f77098d3c602b2e0bc86.
- The places icon set has been switched to a Papirus based one, now you might see yellow folder icons everywhere, changed by https://github.com/linuxmint/mint-y-icons/commit/7f804620de2955d7d91e60ddfa1bdfb1fcd967b0.
- The Mint-Y-Yellow theme is gone, changed by https://github.com/linuxmint/mint-y-icons/commit/41fcdb33cd845861e14ab595d0c27259b50459b9.

If some of your settings are "changed" / "broken" after this update, reconfigure them in the cinnamon-settings app.

### CODE REVIEW

**Comment by mkg20001** [comment]
@bobby285271 for disabling @r-ryantm making prs we'd need to contact @ryantm (hi, could you please disable cinnamon.* for now?)

**Comment by ryantm** [comment]
@mkg20001 Should be skipped now until further notice!

File: pkgs/desktops/cinnamon/cinnamon-common/default.nix
Comment (mkg20001) [inline pkgs/desktops/cinnamon/cinnamon-common/default.nix:152]: why are those getting removed?

Comment (bobby285271) [reply pkgs/desktops/cinnamon/cinnamon-common/default.nix:152]: https://github.com/linuxmint/cinnamon/pull/11244

**Review by mkg20001 — APPROVED** [review]

**Comment by bobby285271** [comment]
Okay the cinnamon tests fails because http://packages.linuxmint.com/pool/main/m/mint-artwork/mint-artwork_1.6.2.tar.xz is 404 and 1.6.3 is out, it looks like upstream only keeps some specific versions :thinking:

**Comment by bobby285271** [comment]
So... here is how the new icons look like :sweat_drops:

**Comment by bobby285271** [comment]
Per [my version checker](https://bobby285271.github.io/what-changed/003-cinnamon.html) all packages with major changes should have at least one new tag now. I plan to merge this bump in this weekend, if you wish to review this PR please do, thanks.

**Comment by dotlambda** [comment]
> rl-23.05: Mention cinnamon 5.6 update

Why don't we include this in 22.11?
cc @NixOS/nixos-release-managers

**Comment by bobby285271** [comment]
:bow: Sorry I didn't consider this as release blocker since this is too recent, and I didn't get this in before branch off. Consider no mutter rebase and gjs rebase are evolved here, do you have specific bug fixes that you want to backport?

**Comment by dotlambda** [comment]
> Since no mutter rebase and gjs rebase are involved here, do you have specific bug fixes that you want to backport?

No, but we could back port the entire PR. I don't use Cinnamon and have no idea how stable 5.6 is, so it's your call.

**Comment by jonringer** [comment]
> No, but we could back port the entire PR. I don't use Cinnamon and have no idea how stable 5.6 is, so it's your call.

The main consideration is if it would "break" any behavior from the previous release. Since desktop platforms usually have a very large scope (high likelihood of regressions), I'm apprehensive about backporting.

**Comment by bobby285271** [comment]
@ofborg test cinnamon
@ofborg build hypnotix timeshift cinnamon.mint-artwork

**Comment by mkg20001** [comment]
@ryantm you can enable  cinnamon.* now

**Comment by bobby285271** [comment]
TBH r-ryantm won't have much chance to open PRs before I retired, I have some scripts that check updates twice a day and I batch everything in a PR and rebuild my laptop to test that at least once a day. You may check how this is done in [`pkgs/desktops/pantheon`](https://github.com/NixOS/nixpkgs/commits/master/pkgs/desktops/pantheon) in the past year :upside_down_face: 

But anyway I guess I want to mention here that I skipped 2 updates here:

- cinnamon 5.6.3: The only change is later reverted.
- warpinator 1.4.0: I guess it has an undefined 'self' in [auth.py](https://github.com/linuxmint/warpinator/blob/9f17136d4490b0089bb82a7f71f51acd0e9fff8b/src/auth.py#L111) that will prevent this from working.

**Comment by ryantm** [comment]
@mkg20001 Re-enabled thanks.

**Pull request merged**
